template.tpl

```diff
@@ -253,9 +262,11 @@ _cnx.push(['referrerUrl', data.referrerUrl.toString()]);
 _cnx.push(['couponApplied', data.couponApplied.toString()]);
 _cnx.push(['pageId', data.pageId.toString()]); 
 _cnx.push(['segment', data.segment.toString()]);
-_cnx.push(['cart', data.cart]); 
+_cnx.push(['spa', data.spa]);
+_cnx.push(['src', 'GTM']);
+_cnx.push(['cart', data.cart]);
 _cnx.push(['customValue1', data.customVal1.toString()]);
-_cnx.push(['customValue2', data.customVal2.toString()]); 
+_cnx.push(['customValue2', data.customVal2.toString()]);
```

Comment:
🟡 _(take it or leave it)_: it might be better to flip the `data` initialization and `_cnx` instantiation to be more declarative as such, here we are reducing mutability that may result in unintended side effects: 

```
for (const key in data) {
  if(!data[key]) {
    data[key] = ''; 
  }
}

const _cnx = [
    ['mid', data.mid.toString()],
    ..., // omitted for brevity
    ['customValue2', data.customVal2.toString()]
]
```

Reply:
In this case, I'd leave _cnx as is because it gets mutated later on in the init file. If we set it to const, we'll cause errors down the line.